include/boost/geometry/strategies/cartesian/closest_points_projected_point.hpp

```diff
@@ -0,0 +1,290 @@
+
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace closest_points
+{
+
+template
+<
+    typename CalculationType = void,
+    typename Strategy = distance::pythagoras<CalculationType>
+>
+class projected_point
+{
+public :
+
+    typedef within::cartesian_point_point equals_point_point_strategy_type;
+
+    typedef intersection::cartesian_segments
+        <
+            CalculationType
+        > relate_segment_segment_strategy_type;
+
+    static inline relate_segment_segment_strategy_type get_relate_segment_segment_strategy()
+    {
+        return relate_segment_segment_strategy_type();
+    }
+
+    typedef within::cartesian_winding
+        <
+            void, void, CalculationType
+        > point_in_geometry_strategy_type;
+
+    static inline point_in_geometry_strategy_type get_point_in_geometry_strategy()
+    {
+        return point_in_geometry_strategy_type();
+    }
+
+    // Integer coordinates can still result in FP distances.
+    // There is a division, which must be represented in FP.
+    // So promote.
+    template <typename Point, typename PointOfSegment>
+    struct calculation_type
+        : promote_floating_point
+          <
+              typename strategy::distance::services::return_type
+                  <
+                      Strategy,
+                      Point,
+                      PointOfSegment
+                  >::type
+          >
+    {};
+
+    template <typename Point, typename PointOfSegment>
+    struct closest_point_result
+    {
+        typedef geometry::detail::closest_points::result
+                <
+                    typename calculation_type<Point, PointOfSegment>::type
+                > type;
+    };
+
+    template <typename Point, typename PointOfSegment>
+    inline typename closest_point_result<Point, PointOfSegment>::type
+    apply(Point const& p, PointOfSegment const& p1, PointOfSegment const& p2) const
+    {
+        assert_dimension_equal<Point, PointOfSegment>();
+
+        typename closest_point_result<Point, PointOfSegment>::type
+                 closest_point_result;
+
+        typedef typename calculation_type<Point, PointOfSegment>::type CT;
+
+        typedef formula::comparable_cartesian_point_segment_distance<CT, true>
+                cartesian_ps_distance;
+        typename cartesian_ps_distance::result_type result
+                = cartesian_ps_distance::apply(p, p1, p2,
+                         distance::services::get_comparable<Strategy>
+                                 ::apply(Strategy()));
+
+        closest_point_result.lon1 = get<0>(p);
+        closest_point_result.lat1 = get<1>(p);
+        closest_point_result.lon2 = result.x2;
+        closest_point_result.lat2 = result.y2;
+        closest_point_result.distance = result.distance;
+
+        return closest_point_result;
+    }
+
+    template <typename ResultType>
+    ResultType
+    apply(ResultType comparable_result) const
```

Comment:
Here the line wouldn't be too long. Would it?

Reply:
no, changed.